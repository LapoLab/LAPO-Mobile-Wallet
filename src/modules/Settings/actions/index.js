// @flow
import { createActions } from 'redux-actions';

export const { recoveryWalletSetting, resyncWalletSetting, printPublicAddress } = createActions({
  RECOVERY_WALLET_SETTING: (data = '') => data,
  RESYNC_WALLET_SETTING: (data = '') => data,
  PRINT_PUBLIC_ADDRESS: (data = '') => data,
});
