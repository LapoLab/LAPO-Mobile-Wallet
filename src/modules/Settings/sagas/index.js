// @flow
import recoveryWallet from './recoveryWallet';
import resyncWallet from './resyncWallet';
import printPublicAddress from './printPublicAddress';

export * from '../actions';

export default {
  recoveryWallet,
  resyncWallet,
  printPublicAddress,
};
