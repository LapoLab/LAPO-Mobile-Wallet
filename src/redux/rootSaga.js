import { all } from 'redux-saga/effects';
import commonsSagas from '../commons/sagas';
import recoveryWordsSagas from '../modules/RecoveryWords/sagas';
import sendSagas from '../modules/Send/sagas';
import paperWordsConfirm from '../modules/PaperWordsConfirm/sagas';
import dashboard from '../modules/Dashboard/sagas';
import settings from '../modules/Settings/sagas';
import paperWords from '../modules/PaperWords/sagas';
import transactionsDetails from '../modules/TransactionDetails/sagas';
import chooseContact from '../modules/ChooseContact/sagas';
import splashScreen from '../modules/SplashScreen/sagas';
import switchContacts from '../modules/SwitchContacts/sagas';
import receive from '../modules/Receive/sagas';

// eslint-disable-next-line
const getListeners = (...args) => args.reduce((acc, nextArg) => [...acc, ...Object.values(nextArg).map(func => func())], []);

export default function* rootSaga() {
  yield all(
    getListeners(
      commonsSagas,
      recoveryWordsSagas,
      sendSagas,
      paperWordsConfirm,
      dashboard,
      settings,
      paperWords,
      transactionsDetails,
      chooseContact,
      splashScreen,
      switchContacts,
      receive,
    ),
  );
}
