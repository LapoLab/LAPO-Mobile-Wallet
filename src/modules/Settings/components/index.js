/**
 * @flow
 */

import React, { Component } from 'react';
import {
  Text,
  View,
  Image,
  TouchableOpacity,
  SafeAreaView,
  BackHandler,
} from 'react-native';
import { NavigationActions } from 'react-navigation';
import LinearGradient from 'react-native-linear-gradient';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import SpinnerPage from '../../../commons/components/SpinnerPage';
import {
  setConfirmPinsRecovery as setConfirmPinsRecoveryAction,
  setPinsRecovery as setPinsRecoveryAction,
  share12Words as share12WordsAction,
  setIsRecovery as setIsRecoveryAction,
  setShowSpinner as setShowSpinnerAction,
} from '../../../commons/actions';
import * as actions from '../actions';
import * as actionsAlert from '../../Alert/actions';

import styles from './styles';
import {
  backIcon, restoreIcon, resetIcon, printPaperWordsIcon, backupIcon,
} from '../../../assets';

type Props = {
  navigation: { openDrawer: Function, dispatch: Function, goBack: Function },
  share12Words: Function,
  recoveryWalletSetting: Function,
  printPublicAddress: Function,
  resyncWalletSetting: Function,
  setIsRecovery: Function,
  isOpenDrawer: boolean,
  showAlert: Function,
  hideAlert: Function,
};
class Settings extends Component<Props> {
  componentDidMount() {
    BackHandler.addEventListener('hardwareBackPress', this.handleBack);
  }

  componentWillUnmount() {
    BackHandler.removeEventListener('hardwareBackPress', this.handleBack);
  }

  handleBack = async () => {
    const { isOpenDrawer } = this.props;
    if (isOpenDrawer) {
      this.navigateToScreen('Dashboard')();
      return true;
    }
    this.handleGoToBack();
    return true;
  };

  navigateToScreen = route => () => {
    const {
      navigation: { dispatch },
    } = this.props;
    const navigateAction = NavigationActions.navigate({
      routeName: route,
    });
    dispatch(navigateAction);
  };

  handleBackup = () => {
    const { share12Words } = this.props;
    share12Words();
  };


  handleGoToBack = () => {
    const { navigation: { openDrawer }, setIsRecovery } = this.props;
    setIsRecovery(false);
    openDrawer();
  }

  handleSavePdf = async () => {
    const { printPublicAddress } = this.props;
    printPublicAddress();
  };

  handleRecoveryWallet = async () => {
    const { recoveryWalletSetting } = this.props;
    recoveryWalletSetting();
  };

  handleResync = () => {
    const { resyncWalletSetting, showAlert, hideAlert } = this.props;

    showAlert(
      'Message',
      'Do you want to resync blocks headers? It may take a long time',
      [
        {
          text: 'OK',
          onPress: () => {
            resyncWalletSetting();
            hideAlert();
          },
        },
        {
          text: 'Cancel',
        },
      ],
      {
        cancelable: true,
      },
    );
  };

  render() {
    return (
      <LinearGradient
        start={{ x: -0.1, y: -0.1 }}
        end={{ x: 1.2, y: 1.2 }}
        locations={[0.04, 0.96]}
        colors={['#4183BD', '#093B0C']}
        style={{ flex: 1 }}
      >
        <SafeAreaView style={styles.container}>
          <View style={styles.header}>
            <TouchableOpacity onPress={this.handleGoToBack}>
              <Image style={styles.backHeader} source={backIcon} />
            </TouchableOpacity>

            <Text style={styles.titleHeader}>App settings</Text>
          </View>

          <View style={styles.links}>
            <TouchableOpacity style={styles.link} onPress={this.handleRecoveryWallet}>
              <Image style={styles.icon} source={restoreIcon} />
              <View style={styles.info}>
                <Text style={styles.title}>Restore wallet</Text>
                <Text style={styles.subTitle}>Restore a wallet that you previously created</Text>
              </View>
            </TouchableOpacity>

            <View style={styles.line} />

            <TouchableOpacity style={styles.link} onPress={this.handleBackup}>
              <Image style={styles.icon} source={backupIcon} />
              <View style={styles.info}>
                <Text style={styles.title}>Backup wallet</Text>
                <Text style={styles.subTitle}>Backup your wallet</Text>
              </View>
            </TouchableOpacity>

            <View style={styles.line} />

            <TouchableOpacity style={styles.link} onPress={this.handleResync}>
              <Image style={styles.icon} source={resetIcon} />
              <View style={styles.info}>
                <Text style={styles.title}>Reset the blockchain</Text>
                <Text style={styles.subTitle}>Reset the blockchain data stored locally</Text>
              </View>
            </TouchableOpacity>

            <View style={styles.line} />

            <TouchableOpacity style={styles.link} onPress={this.handleSavePdf}>
              <Image style={styles.icon} source={printPaperWordsIcon} />
              <View style={styles.info}>
                <Text style={styles.title}>Print paper wallet</Text>
                <Text style={styles.subTitle}>Save in PDF or print your paper wallet</Text>
              </View>
            </TouchableOpacity>
          </View>
        </SafeAreaView>
        <SpinnerPage />
      </LinearGradient>
    );
  }
}

const mapStateToProps = state => ({
  pins: state.pins.pins,
  paperWords: state.wallet.paperWords,
  isOpenDrawer: state.modals.isOpenDrawer,
});

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    setConfirmPins: setConfirmPinsRecoveryAction,
    setPins: setPinsRecoveryAction,
    setIsRecovery: setIsRecoveryAction,
    setShowSpinner: setShowSpinnerAction,
    share12Words: share12WordsAction,
    recoveryWalletSetting: actions.recoveryWalletSetting,
    resyncWalletSetting: actions.resyncWalletSetting,
    printPublicAddress: actions.printPublicAddress,
    showAlert: actionsAlert.showAlert,
    hideAlert: actionsAlert.hideAlert,
  },
  dispatch,
);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Settings);
