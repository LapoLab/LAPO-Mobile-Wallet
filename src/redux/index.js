import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import rootReducer from './rootReducer';
import rootSagas from './rootSaga';

const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
  blacklist: [
    'modals',
    'dashboard',
    'paperWords',
    'chooseContact',
    'send',
    'transactionsDetails',
    'alert',
    'pinsRecovery',
  ],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const sagaMiddleware = createSagaMiddleware();
// eslint-disable-next-line no-undef
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(sagaMiddleware)),
);
sagaMiddleware.run(rootSagas);

export const persistor = persistStore(store);
