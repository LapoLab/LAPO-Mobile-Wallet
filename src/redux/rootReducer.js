import { combineReducers } from 'redux';

import wallet from '../commons/reducers/wallet';
import pins from '../commons/reducers/pins';
import pinsRecovery from '../commons/reducers/pinsRecovery';
import modals from '../commons/reducers/modals';
import settings from '../commons/reducers/settings';
import notify from '../modules/Notify/reducers';
import paperConfirm from '../modules/PaperWordsConfirm/reducer';
import dashboard from '../modules/Dashboard/reducers';
import paperWords from '../modules/PaperWords/reducers';
import chooseContact from '../modules/ChooseContact/reducers';
import send from '../modules/Send/reducers';
import transactionsDetails from '../modules/TransactionDetails/reducer';
import alert from '../modules/Alert/reducers';

export default combineReducers({
  wallet,
  pins,
  pinsRecovery,
  notify,
  modals,
  settings,
  paperConfirm,
  dashboard,
  paperWords,
  chooseContact,
  send,
  transactionsDetails,
  alert,
});
