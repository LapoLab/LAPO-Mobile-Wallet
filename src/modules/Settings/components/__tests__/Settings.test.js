// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../redux';
import Settings from '../index';

describe('Settings page', () => {
  test('renders snapshots', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Settings />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  }); // eslint-disable-line
});
