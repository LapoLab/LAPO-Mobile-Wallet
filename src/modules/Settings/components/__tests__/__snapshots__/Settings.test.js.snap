// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings page renders snapshots 1`] = `
<BVLinearGradient
  colors={
    Array [
      -12483651,
      -16172276,
    ]
  }
  endPoint={
    Object {
      "x": 1.2,
      "y": 1.2,
    }
  }
  locations={
    Array [
      0.04,
      0.96,
    ]
  }
  startPoint={
    Object {
      "x": -0.1,
      "y": -0.1,
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "flex-start",
          "marginTop": 20,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/back.png",
            }
          }
          style={
            Object {
              "flex": 1,
              "marginLeft": 12,
              "resizeMode": "contain",
              "width": 12,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "white",
            "flex": 1,
            "fontFamily": "Lato-Bold",
            "fontSize": 18,
            "marginRight": 24,
            "textAlign": "center",
          }
        }
      >
        App settings
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "justifyContent": "center",
          "marginTop": 50,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 100,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/restore.png",
            }
          }
          style={
            Object {
              "height": 48,
              "resizeMode": "contain",
              "width": 100,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Lato-Bold",
                "fontSize": 20,
              }
            }
          >
            Restore wallet
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Lato-Light",
                "fontSize": 16,
                "marginTop": 5,
              }
            }
          >
            Restore a wallet that you previously created
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderTopColor": "#C9C9C9",
            "borderTopWidth": 1,
            "width": "100%",
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 100,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/backupIcon.png",
            }
          }
          style={
            Object {
              "height": 48,
              "resizeMode": "contain",
              "width": 100,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Lato-Bold",
                "fontSize": 20,
              }
            }
          >
            Backup wallet
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Lato-Light",
                "fontSize": 16,
                "marginTop": 5,
              }
            }
          >
            Backup your wallet
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderTopColor": "#C9C9C9",
            "borderTopWidth": 1,
            "width": "100%",
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 100,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/reset.png",
            }
          }
          style={
            Object {
              "height": 48,
              "resizeMode": "contain",
              "width": 100,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Lato-Bold",
                "fontSize": 20,
              }
            }
          >
            Reset the blockchain
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Lato-Light",
                "fontSize": 16,
                "marginTop": 5,
              }
            }
          >
            Reset the blockchain data stored locally
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderTopColor": "#C9C9C9",
            "borderTopWidth": 1,
            "width": "100%",
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 100,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/printPaperWords.png",
            }
          }
          style={
            Object {
              "height": 48,
              "resizeMode": "contain",
              "width": 100,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Lato-Bold",
                "fontSize": 20,
              }
            }
          >
            Print paper wallet
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Lato-Light",
                "fontSize": 16,
                "marginTop": 5,
              }
            }
          >
            Save in PDF or print your paper wallet
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
